consensus/wal.go

```diff
@@ -73,13 +73,13 @@ type baseWAL struct {
 	enc *WALEncoder
 }
 
-func NewWAL(walFile string) (*baseWAL, error) {
+func NewWAL(walFile string, groupOptions ...func(*auto.Group)) (*baseWAL, error) {
```

Comment:
what are the advantages of options over config struct?

Reply:
if use config struct, all the places use NewWAL need modify to init and add a param.

Reply:
> NewWAL need modify to init

that's fine.

> add a param

we can say that if param == 0, then default value will be applied

but let's leave options